manifests/resource/location.pp

```diff
@@ -153,66 +153,65 @@
 #  }
 
 define nginx::resource::location (
-  $ensure                      = present,
-  $internal                    = false,
-  $location                    = $name,
-  $server                      = undef,
-  $www_root                    = undef,
-  $autoindex                   = undef,
-  $index_files                 = [
+  Enum['present', 'absent'] $ensure                   = present,
+  Boolean $internal                                   = false,
+  String $location                                    = $name,
+  String $server                                      = undef,
```

Comment:
yes, that makes it a required parameter. It currently already is a required param, just not handled nicely:
https://github.com/voxpupuli/puppet-nginx/blob/master/manifests/resource/location.pp#L385-L387
https://github.com/voxpupuli/puppet-nginx/blob/master/manifests/resource/location.pp#L230-L232

It would have been more clear if it didn't default to undef but to nothing.